lib/src/solver/report.dart

```diff
@@ -199,9 +207,13 @@ $contentHashesDocumentationUrl
       for (var name in ordered(removed)) {
         await _reportPackage(name, alwaysShow: true);
       }
+      hasChanges = true;
     }
 
     log.message(_output);
+    if (enforceLockfile && hasChanges) {
+      dataError('Could not enforce the lockfile.');
```

Comment:
Would it not be better if these error messages mentioned:
 * Resolution aborted: Unable to satisfy `pubspec.yaml` using the existing `pubspec.lock`
 * Try without `--enforce-lockfile` to update `pubspec.lock`.
 * X number of dependencies changed in `pubspec.lock`.
 * `pubspec.lock` does not exist (if that is the case).

Just some ideas for information that should go into this message.

---------------------------------

Interesting side note:

It's unclear if we actually should allow resolution to go ahead. I think it's the best implementation for us.
But it leaves the user with weird behavior, where `--enforce-lockfile` can cause a dependency conflict.
Which should never be possible, because `--enforce-lockfile` implies that I don't want resolution to be taking place at all.

But I'm not sure we want the number of hacks necessary to work around this.

Reply:
> X number of dependencies changed in pubspec.lock.

This is mentioned by the report.summary.

> pubspec.lock does not exist (if that is the case).

That case is handled separately before even attempting to resolve.